custom_components/pyscript/__init__.py

```diff
@@ -250,6 +266,55 @@ async def unload_scripts(global_ctx_only=None, unload_all=False):
         await GlobalContextMgr.delete(global_ctx_name)
 
 
+@bind_hass
+async def install_requirements(hass):
+    """Install missing requirements from requirements.txt."""
+    requirements_path = os.path.join(hass.config.path(FOLDER), "requirements.txt")
+
+    if os.path.exists(requirements_path):
+        with open(requirements_path, "r") as requirements_file:
+            requirements_to_install = []
+            for pkg in requirements_file.readlines():
+                # Remove inline comments which are accepted by pip but not by Home
+                # Assistant's installation method.
+                # https://rosettacode.org/wiki/Strip_comments_from_a_string#Python
+                i = pkg.find("#")
+                if i >= 0:
+                    pkg = pkg[:i].strip()
+
+                try:
+                    # Attempt to get version of package. Do nothing if it's found since
+                    # we want to use the version that's already installed to be safe
+                    requirement = pkg_resources.Requirement.parse(pkg)
+                    requirement_installed_version = version(requirement.project_name)
+
+                    if requirement_installed_version in requirement:
+                        _LOGGER.debug("`%s` already found", requirement.project_name)
+                    else:
+                        _LOGGER.debug(
```

Comment:
This should probably be a `LOGGER.warning` level message, since it's important for the user to know if there is a version mismatch that is being ignored.